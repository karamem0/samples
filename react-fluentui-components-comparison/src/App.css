/*
* Copyright (c) 2011-2024 karamem0
*
* This software is released under the MIT License.
*
* https://github.com/karamem0/samples/blob/main/LICENSE
*/

.app {
  display: flex;
  align-items: center;
  justify-content: center;
}

.table th:first-child {
  width: 5rem;
}

.table th {
  width: 15rem;
}

.table td {
  padding: 0.25rem;
}

.table td > div {
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 0.25rem;
}
