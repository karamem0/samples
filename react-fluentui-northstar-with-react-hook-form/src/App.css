/*
* Copyright (c) 2011-2024 karamem0
*
* This software is released under the MIT License.
*
* https://github.com/karamem0/samples/blob/main/LICENSE
*/

.app {
  display: grid;
  row-gap: 1rem;
  padding: 1rem;
}
