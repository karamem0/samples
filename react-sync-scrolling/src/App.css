/*
* Copyright (c) 2011-2024 karamem0
*
* This software is released under the MIT License.
*
* https://github.com/karamem0/samples/blob/main/LICENSE
*/

html, body {
  margin: 0;
  height: 100vh;
}

#root {
  padding: 1rem;
}

.container {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 1fr;
  gap: 1rem;
  height: calc(100vh - 2rem);
}
