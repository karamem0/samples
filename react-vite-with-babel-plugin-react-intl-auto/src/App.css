/*
* Copyright (c) 2011-2024 karamem0
*
* This software is released under the MIT License.
*
* https://github.com/karamem0/samples/blob/main/LICENSE
*/

.app {
  text-align: center;
}

.app-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .app-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.app-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.app-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

button {
  font-size: calc(10px + 2vmin);
}
