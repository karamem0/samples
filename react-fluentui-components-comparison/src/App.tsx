//
// Copyright (c) 2011-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/samples/blob/main/LICENSE
//

import React from 'react';
import * as V8 from '@fluentui/react';
import * as V0 from '@fluentui/react-northstar';
import * as V9 from '@fluentui/react-components';
import * as V9Unstable from '@fluentui/react-components/unstable';
import * as MUI from '@material-ui/core';
import './App.css';

V8.initializeIcons();

function App() {

  return (
    <div className="app">
      <V0.Provider theme={V0.teamsTheme}>
        <V9.FluentProvider theme={V9.webLightTheme}>
          <div>
            <header>
              <h1>Fluent UI Components Comparison</h1>
            </header>
            <table className="table">
              <thead>
                <tr>
                  <th>Name</th>
                  <th>@fluentui/react (v8)</th>
                  <th>@fluentui/react-northstar (v0)</th>
                  <th>@fluentui/react-components (v9)</th>
                  <th>@material-ui/core (v4)</th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>Button</td>
                  <td>
                    <div>
                      <V8.DefaultButton>Cancel</V8.DefaultButton>
                      <V8.PrimaryButton>OK</V8.PrimaryButton>
                    </div>
                  </td>
                  <td>
                    <div>
                      <V0.Button>Cancel</V0.Button>
                      <V0.Button primary>OK</V0.Button>
                    </div>
                  </td>
                  <td>
                    <div>
                      <V9.Button>Cancel</V9.Button>
                      <V9.Button appearance="primary">OK</V9.Button>
                    </div>
                  </td>
                  <td>
                    <div>
                      <MUI.Button variant="contained">Cancel</MUI.Button>
                      <MUI.Button color="primary" variant="contained">OK</MUI.Button>
                    </div>
                  </td>
                </tr>
                <tr>
                  <td>Text Input</td>
                  <td>
                    <div>
                      <V8.TextField placeholder="Title" />
                    </div>
                  </td>
                  <td>
                    <div>
                      <V0.Input placeholder="Title" />
                    </div>
                  </td>
                  <td>
                    <div>
                      <V9.Input placeholder="Title" />
                    </div>
                  </td>
                  <td>
                    <div>
                      <MUI.TextField
                        label="Title"
                        placeholder="Title"
                        variant="outlined" />
                    </div>
                  </td>
                </tr>
                <tr>
                  <td>Radio Button</td>
                  <td>
                    <div>
                      <V8.ChoiceGroup options={[{ key: '1', text: 'Windows' }]} />
                    </div>
                  </td>
                  <td>
                    <div>
                      <V0.RadioGroupItem label="Windows" />
                    </div>
                  </td>
                  <td>
                    <div>
                      <V9.Radio label="Windows" />
                    </div>
                  </td>
                  <td>
                    <div>
                      <MUI.FormControlLabel
                        control={<MUI.Radio color="primary" />}
                        label="Windows" />
                    </div>
                  </td>
                </tr>
                <tr>
                  <td>Checkbox</td>
                  <td>
                    <div>
                      <V8.Checkbox label="Mark as read" />
                    </div>
                  </td>
                  <td>
                    <div>
                      <V0.Checkbox label="Mark as read" />
                    </div>
                  </td>
                  <td>
                    <div>
                      <V9.Checkbox label="Mark as read" />
                    </div>
                  </td>
                  <td>
                    <div>
                      <MUI.FormControlLabel
                        control={<MUI.Checkbox color="primary" />}
                        label="Mark as read" />
                    </div>
                  </td>
                </tr>
                <tr>
                  <td>Toggle</td>
                  <td>
                    <div>
                      <V8.Toggle inlineLabel label="Mark as read" />
                    </div>
                  </td>
                  <td>
                    <div>
                      <V0.Checkbox toggle label="Mark as read" />
                    </div>
                  </td>
                  <td>
                    <div>
                      <V9.ToggleButton>Mark as read</V9.ToggleButton>
                    </div>
                  </td>
                  <td>
                    <div>
                      <MUI.FormControlLabel
                        control={<MUI.Switch color="primary" />}
                        label="Mark as read" />
                    </div>
                  </td>
                </tr>
                <tr>
                  <td>Dropdown</td>
                  <td>
                    <div>
                      <V8.Dropdown
                        styles={{ dropdown: { width: '10rem' } }}
                        options={[{ key: '1', text: 'Windows' }]} />
                    </div>
                  </td>
                  <td>
                    <div>
                      <V0.Dropdown items={[{ header: 'Windows' }]} fluid />
                    </div>
                  </td>
                  <td>
                    <div>
                      <V9Unstable.Dropdown>
                        <V9Unstable.Option>Windows</V9Unstable.Option>
                      </V9Unstable.Dropdown>
                    </div>
                  </td>
                  <td>
                    <div>
                      <MUI.FormControl variant="outlined">
                        <MUI.InputLabel id="mui-select">Windows</MUI.InputLabel>
                        <MUI.Select
                          labelId="mui-select"
                          label="Windows"
                          style={{ width: '10rem' }}
                          variant="outlined">
                          <MUI.MenuItem value="Windows">Windows</MUI.MenuItem>
                        </MUI.Select>
                      </MUI.FormControl>
                    </div>
                  </td>
                </tr>
                <tr>
                  <td>Link</td>
                  <td>
                    <div>
                      <V8.Link>GitHub</V8.Link>
                    </div>
                  </td>
                  <td>
                    <div>
                      - (N/A)
                    </div>
                  </td>
                  <td>
                    <div>
                      <V9.Link>GitHub</V9.Link>
                    </div>
                  </td>
                  <td>
                    <div>
                      <MUI.Link>GitHub</MUI.Link>
                    </div>
                  </td>
                </tr>
                <tr>
                  <td>Date Picker</td>
                  <td>
                    <div>
                      <V8.DatePicker style={{ width: '10rem' }} />
                    </div>
                  </td>
                  <td>
                    <div>
                      <V0.Datepicker />
                    </div>
                  </td>
                  <td>
                    <div>
                      - (N/A)
                    </div>
                  </td>
                  <td>
                    <div>
                      <MUI.TextField type="date" />
                    </div>
                  </td>
                </tr>
                <tr>
                  <td>Loader</td>
                  <td>
                    <div>
                      <V8.Spinner />
                    </div>
                  </td>
                  <td>
                    <div>
                      <V0.Loader />
                    </div>
                  </td>
                  <td>
                    <div>
                      <V9.Spinner />
                    </div>
                  </td>
                  <td>
                    <div>
                      <MUI.CircularProgress />
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </V9.FluentProvider>
      </V0.Provider>
    </div>
  )
}

export default App;
