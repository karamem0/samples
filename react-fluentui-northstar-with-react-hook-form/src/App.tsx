//
// Copyright (c) 2011-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/samples/blob/main/LICENSE
//

import React from 'react';
import './App.css';
import {
  Card,
  Form,
  FormButton,
  FormDatepicker,
  FormInput,
  FormMessage,
  FormTextArea,
  Provider,
  teamsTheme
} from '@fluentui/react-northstar';
import {
  Controller,
  FieldErrors,
  useForm
} from 'react-hook-form';

interface IDiaryForm {
  date?: Date,
  title?: string,
  content?: string,
}

function App() {
  const form = useForm<IDiaryForm>();
  const [json, setJson] = React.useState<string>("");

  const handleSubmitValid = React.useMemo(() => (value: IDiaryForm) => {
    setJson(JSON.stringify(value));
  }, []);

  const handleSubmitError = React.useMemo(() => (error: FieldErrors) => {
    console.error(error);
  }, []);

  return (
    <Provider theme={teamsTheme}>
      <div className="app">
        <Form onSubmit={form.handleSubmit(handleSubmitValid, handleSubmitError)}>
          <Controller
            control={form.control}
            defaultValue={new Date()}
            name="date"
            rules={{
              required: true,
            }}
            render={({ field, fieldState }) => (
              <FormDatepicker
                ref={field.ref}
                errorMessage={fieldState.error?.message}
                label="Date"
                selectedDate={field.value}
                onDateChangeError={(_, value) => form.setError(field.name, { message: value?.error })}
                onDateChange={(_, props) => field.onChange(props?.value)}
                onBlur={field.onBlur}
              />
            )}
          />
          <Controller
            control={form.control}
            defaultValue={""}
            name="title"
            rules={{
              required: 'The value is requred',
            }}
            render={({ field, fieldState }) => (
              <FormInput
                ref={field.ref}
                errorMessage={fieldState.error?.message}
                fluid
                label="Title"
                placeholder="(Title)"
                value={field.value}
                onChange={(_, props) => field.onChange(props?.value)}
                onBlur={field.onBlur}
              />
            )}
          />
          <Controller
            control={form.control}
            defaultValue={""}
            name="content"
            rules={{
              required: 'The value is requred',
            }}
            render={({ field, fieldState }) => (
              <FormTextArea
                ref={field.ref}
                errorMessage={fieldState.error?.message}
                fluid
                label="Content"
                placeholder="(Content)"
                value={field.value}
                onChange={(_, props) => field.onChange(props?.value)}
                onBlur={field.onBlur}
              />
            )}
          />
          <FormButton content="Submit" />
        </Form>
        {json && (<Card fluid>{json}</Card>)}
      </div>
    </Provider>
  )
}

export default App
