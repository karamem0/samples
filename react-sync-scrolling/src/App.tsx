//
// Copyright (c) 2011-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/samples/blob/main/LICENSE
//

import React from 'react';
import './App.css'
import ScrollSynchronizer from './ScrollSynchronizer';

function App() {

  return (
    <div className="container">
      <ScrollSynchronizer>
        {
          ({ ref1, ref2 }) => (
            <React.Fragment>
              <textarea ref={ref1} className="textarea1"></textarea>
              <textarea ref={ref2} className="textarea2"></textarea>
            </React.Fragment>
          )
        }
      </ScrollSynchronizer>
    </div>
  );

}

export default App
