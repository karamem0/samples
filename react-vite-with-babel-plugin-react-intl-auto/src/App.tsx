//
// Copyright (c) 2011-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/samples/blob/main/LICENSE
//

import React, { useState } from 'react';
import logo from './logo.svg';
import './App.css';
import { FormattedMessage, IntlProvider } from 'react-intl';
import translations from './translations';
import messages from './messages';

function App() {

  const [count, setCount] = useState(0);

  return (
    <div className="app">
      <header className="app-header">
        <img src={logo} className="app-logo" alt="logo" />
        <p>Hello Vite + React!</p>
        <p>
          <button type="button" onClick={() => setCount((count) => count + 1)}>
            count is: {count}
          </button>
        </p>
        <p>
          Edit <code>App.tsx</code> and save to test HMR updates.
        </p>
        <p>
          <a
            className="app-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
          {' | '}
          <a
            className="app-link"
            href="https://vitejs.dev/guide/features.html"
            target="_blank"
            rel="noopener noreferrer"
          >
            Vite Docs
          </a>
        </p>
        <p>
          <IntlProvider locale='ja' messages={translations.ja}>
            <FormattedMessage {...messages.HelloWorld} />
          </IntlProvider>
        </p>
      </header>
    </div>
  );

}

export default App;
