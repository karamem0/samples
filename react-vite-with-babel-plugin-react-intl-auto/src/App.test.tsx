//
// Copyright (c) 2011-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/samples/blob/main/LICENSE
//

import React from 'react';

import { render, screen } from '@testing-library/react';
import App from './App';

describe('App', () => {

  it('create shapshot', () => {
    render(<App />);
    expect(screen.queryAllByText(/^.*$/)[0]).toMatchSnapshot();
  });

});
