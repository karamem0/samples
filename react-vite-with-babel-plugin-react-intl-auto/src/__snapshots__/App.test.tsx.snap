// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App create shapshot 1`] = `
<body>
  <div>
    <div
      class="app"
    >
      <header
        class="app-header"
      >
        <img
          alt="logo"
          class="app-logo"
          src="[object Object]"
        />
        <p>
          Hello Vite + React!
        </p>
        <p>
          <button
            type="button"
          >
            count is: 
            0
          </button>
        </p>
        <p>
          Edit 
          <code>
            App.tsx
          </code>
           and save to test HMR updates.
        </p>
        <p>
          <a
            class="app-link"
            href="https://reactjs.org"
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn React
          </a>
           | 
          <a
            class="app-link"
            href="https://vitejs.dev/guide/features.html"
            rel="noopener noreferrer"
            target="_blank"
          >
            Vite Docs
          </a>
        </p>
        <p>
          こんにちは世界
        </p>
      </header>
    </div>
  </div>
</body>
`;
